web/packages/teleterm/src/ui/Search/actions.tsx

```diff
@@ -43,13 +43,16 @@ export interface ParametrizedAction {
   searchResult: SearchResult;
   preventAutoClose?: boolean;
   parameter: {
-    getSuggestions(): Promise<string[]>;
+    getSuggestions(): Promise<Parameter[]>;
+    /** Disables providing new values. */
+    allowOnlySuggestions?: boolean;
     placeholder: string;
   };
-
-  perform(parameter: string): void;
+  perform(parameter: Parameter): void;
 }
 
+export type Parameter = { key: string; display: string };
```

Comment:
I'm not sure about `key`, what's the story behind that name?

I was thinking of something like `value` and `displayText`. `value` and `display` would work too.

Reply:
I think it came from React's `key`, but `value` sounds much better.